ibis/expr/operations/relations.py

```diff
@@ -330,6 +340,13 @@ class DatabaseTable(PhysicalTable):
     namespace: Namespace = Namespace()
 
 
+@public
+class VersionedDatabaseTable(DatabaseTable):
```

Comment:
Can other table-like objects be versioned? I'm thinking of views or generic `SELECT` queries here. If yes, it might be better to have a `TableSnapshot` relation instead referencing an arbitrary parent relation, like:

```py
class TableSnapshot(Relation):
    parent: Relation  # this would be an ops.DatabaseTable most of the cases
    at_time: Column[dt.Timestamp]
```

Reply:
I believe we already discussed this, and the answer was no. Even if it turns out to be wrong, it should be addressed in a follow up PR.

Reply:
In my understanding, intermediate tables in Flink such as `SELECT` queries can be versioned and queried only with [Paimon](https://www.ververica.com/blog/apache-paimon-the-streaming-lakehouse) in place. Without `Paimon`, Flink supports [`versioned views`](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/concepts/versioned_tables/#versioned-table-views), against which temporal join can be performed.